system/api/server.go

```diff
@@ -9,4 +9,6 @@ func Run() {
 	http.HandleFunc("/api/content", Record(CORS(Gzip(contentHandler))))
 
 	http.HandleFunc("/api/content/external", Record(CORS(externalContentHandler)))
+
+	http.HandleFunc("/api/content/update", Record(CORS(updateContentHandler)))
```

Comment:
I'd add PATCH as an option here for the updateContentHandler.  All other things being equal, I'd go with that.  My only reservation would be that external and update currently are sent as form values, so moving away from POST maybe you lose some compatibility for forms use cases that only support GET/POST.